Examples/CMakeLists.txt

```diff
@@ -0,0 +1,221 @@
+            ERROR_VARIABLE MPIEXEC_HELP_TEXT
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+        )
+        if(${MPIEXEC_HELP_TEXT} MATCHES "^.*allow-run-as-root.*$")
+            set(MPI_ALLOW_ROOT --allow-run-as-root)
+        endif()
+    endif()
+endif()
+
+# Add a WarpX test set (with sub-tests)
+#
+# name: unique name of this test
+# dims: 1,2,RZ,3
+# nprocs: 1 or 2 (maybe refactor later on to just depend on WarpX_MPI)
+# eb: needs EB support? (temporary until runtime param)
+# inputs: inputs file or PICMI script - WarpX_MPI decides on w/wo MPI
+# analysis: analysis script - always run without MPI
+# output: output file to analyze
+# dependency: name of another test to run first
+#
+function(add_warpx_test
+    name
+    dims
+    nprocs
+    eb
+    inputs
+    analysis
+    output
+    dependency
+)
+    # cannot run MPI tests w/o MPI build
+    if(nprocs GREATER_EQUAL 2 AND NOT WarpX_MPI)
+        message(WARNING "${name}: cannot run MPI tests without MPI build")
+        return()
+    endif()
+
+    # cannot run EB tests w/o EB build
+    if(eb AND NOT WarpX_EB)
+        message(WARNING "${name}: cannot run EB tests without EB build")
+        return()
+    endif()
+
+    # do not run no-EB tests w/ EB build
+    if(NOT eb AND WarpX_EB)
+        return()
+    endif()
+
+    # cannot run tests with unsupported geometry
+    if(NOT dims IN_LIST WarpX_DIMS)
+        return()
+    endif()
+
+    # Python test?
+    set(python OFF)
+    if(inputs MATCHES ".*\.py$")
+        set(python ON)
+    endif()
+
+    # cannot run Python tests w/o Python support
+    if(python AND NOT WarpX_PYTHON)
+        return()
+    endif()
+
+    # set dimension suffix
+    warpx_set_suffix_dims(SD ${dims})
+
+    # make a unique run directory
+    file(MAKE_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${name})
+    set(THIS_WORKING_DIR ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${name})
+
+    get_filename_component(INPUTS_ALL ${WarpX_SOURCE_DIR}/${inputs} REALPATH)
+
+    # separate input file (full path) and optional command-line arguments
+    separate_arguments(INPUTS_FILE_ARGS UNIX_COMMAND PROGRAM "${INPUTS_ALL}")
+
+    # get input file (full path)
+    list(GET INPUTS_FILE_ARGS 0 INPUTS_FILE)
+
+    # get optional command-line arguments
+    list(SUBLIST INPUTS_FILE_ARGS 1 -1 INPUTS_ARGS)
+    list(JOIN INPUTS_FILE_ARGS " " INPUTS_FILE_ARGS)
+
+    # get input file's directory
+    get_filename_component(INPUTS_DIR ${INPUTS_FILE} DIRECTORY)
+
+    # copy base input files to run directory
+    file(GLOB INPUTS_BASE "${INPUTS_DIR}/inputs_base_${SD}*")
+    file(COPY ${INPUTS_BASE} DESTINATION ${THIS_WORKING_DIR})
+
+    # copy test input file to run directory
+    file(COPY ${INPUTS_FILE} DESTINATION ${THIS_WORKING_DIR})
+
+    # for restart tests, copy input file of original test (without "_restart" suffix)
+    string(REPLACE "_restart" "" INPUTS_FILE_NORESTART ${INPUTS_FILE})
+    file(COPY ${INPUTS_FILE_NORESTART} DESTINATION ${THIS_WORKING_DIR})
```

Comment:
Discussed today: eventually, we want to remove this convention and just depend on `base` inputs and dependent tests (`dependency`).

Reply:
Just a note to keep the conversation going. 

We can discuss this, but I do think that enforcing the convention (currently already implemented in the new automated style checks - which, by the way, I rewrote as a single Python script `check_inputs.py` in this PR) has advantages.

Right now one can type
```
ctest --test-dir build/ --output-on-failure -R test_3d_acceleration_psatd
```
and get this output, which I think is pretty easy to parse:
```
Internal ctest changing into directory: /home/edoardo/src/warpx/build
Test project /home/edoardo/src/warpx/build
      Start 506: test_3d_acceleration_psatd.run
 1/12 Test #506: test_3d_acceleration_psatd.run .........................   Passed    3.69 sec
      Start 507: test_3d_acceleration_psatd.analysis
 2/12 Test #507: test_3d_acceleration_psatd.analysis ....................   Passed    2.27 sec
      Start 509: test_3d_acceleration_psatd_restart.run
 3/12 Test #509: test_3d_acceleration_psatd_restart.run .................   Passed    2.39 sec
      Start 510: test_3d_acceleration_psatd_restart.analysis
 4/12 Test #510: test_3d_acceleration_psatd_restart.analysis ............   Passed    2.32 sec
      Start 511: test_3d_acceleration_psatd_restart.cleanup
 5/12 Test #511: test_3d_acceleration_psatd_restart.cleanup .............   Passed    0.06 sec
      Start 508: test_3d_acceleration_psatd.cleanup
 6/12 Test #508: test_3d_acceleration_psatd.cleanup .....................   Passed    0.07 sec
      Start 512: test_3d_acceleration_psatd_time_avg.run
 7/12 Test #512: test_3d_acceleration_psatd_time_avg.run ................   Passed    4.88 sec
      Start 513: test_3d_acceleration_psatd_time_avg.analysis
 8/12 Test #513: test_3d_acceleration_psatd_time_avg.analysis ...........   Passed    2.25 sec
      Start 515: test_3d_acceleration_psatd_time_avg_restart.run
 9/12 Test #515: test_3d_acceleration_psatd_time_avg_restart.run ........   Passed    2.94 sec
      Start 516: test_3d_acceleration_psatd_time_avg_restart.analysis
10/12 Test #516: test_3d_acceleration_psatd_time_avg_restart.analysis ...   Passed    2.24 sec
      Start 517: test_3d_acceleration_psatd_time_avg_restart.cleanup
11/12 Test #517: test_3d_acceleration_psatd_time_avg_restart.cleanup ....   Passed    0.06 sec
      Start 514: test_3d_acceleration_psatd_time_avg.cleanup
12/12 Test #514: test_3d_acceleration_psatd_time_avg.cleanup ............   Passed    0.11 sec

100% tests passed, 0 tests failed out of 12
```

If we allow arbitrary names for the restart test based on a given test (e.g., with a "restart" keyword anywhere in the test name - `test_3d_restart_acceleration_psatd` instead of `test_3d_acceleration_psatd_restart`), I think this output will be less easy to parse and/or work with (when it comes to parsing visually, parsing through code, sorting alphabetically, filtering tests in CTest via `-R`, etc.).

The convention does not seem like a big burden to me, the automated style checks can catch instances where it is not followed and reviewers/maintainers can make sure that such instances are fixed - it doesn't have to be an extra task on the developer's end.